src/middlewared/middlewared/plugins/kubernetes_linux/infra_migrations.py

```diff
@@ -0,0 +1,83 @@
+import asyncio
+import json
+import os
+
+from middlewared.service import CallError, Service
+from middlewared.utils.plugins import load_modules
+from middlewared.utils.python import get_middlewared_dir
+
+
+def load_migrations():
+    return sorted(load_modules(
+        os.path.join(get_middlewared_dir(), 'plugins/kubernetes_linux/migrations')
+    ), key=lambda x: x.__name__)
+
+
+class KubernetesMigrationsService(Service):
+
+    MIGRATIONS_FILE_NAME = 'migrations.json'
+
+    class Config:
+        namespace = 'k8s.migration'
+        private = True
+
+    @property
+    def migration_file_path(self):
+        return os.path.join(
+            '/mnt', self.middleware.call_sync('kubernetes.config')['dataset'], self.MIGRATIONS_FILE_NAME
+        )
+
+    def applied(self):
+        try:
+            with open(self.migration_file_path, 'r') as f:
+                return json.loads(f.read())
+        except FileNotFoundError:
+            self.logger.error('%r migration file not found, creating one', self.migration_file_path)
```

Comment:
Isn't it normal that the file does not exist?

Reply:
It won't for users who are going to migrate to the newer version of scale

Reply:
But it's normal for those who already were using apps on previous versions of SCALE

Reply:
Nope it won't as migrations for k8s did not exist at that point. Ah i see, you are pointing out that we should not log this as an error in that case ?

That's tricky as it's possible say users upgraded to newer scale version which has migrations but then they select an apps pool which was used on previous version of scale, now that won't contain the migration file. Should we maybe log this as `warning` instead ?

Reply:
> users upgraded to newer scale version which has migrations but then they select an apps pool which was used on previous version of scale

That's not even a warning, that's a normal situation which will happen to everyone upgrading to 22.12. I don't think we should log everything at all; we'll see `info` messages executing every existing migration below.